### PR TITLE
Correctly handle comments in attributes in doctests source code

### LINKED ISSUE #134221: Adding a comment next to `#![feature(ptr_metadata,)]` silences failing doctests
### Reproduce

https://github.com/ink-feather-org/trait_cast_rs/tree/80a0a0994dfdbf5e8b6978879ce3205188401df1

#### 1:

```shell
git clone https://github.com/ink-feather-org/trait_cast_rs.git
git checkout 80a0a0994dfdbf5e8b6978879ce3205188401df1
cargo test --all-features
```

All tests pass!

#### 2:

Remove the comment in line 43 of the `README.md`:

```rust
#![feature(
  ptr_metadata, // <- Remove this comment
)]
use trait_cast_rs::{
  make_trait_castable, TraitcastableAny, TraitcastableAnyInfra, TraitcastableAnyInfraExt,
};
#[make_trait_castable(Print)]
struct Source(i32);
...
```

Next run the tests again:

```shell
cargo test --all-features
```

Oh no! The test fails with `E0119`.

What is going on?!? I did not expect the comment breaking anything.

### Meta
`rustc --version --verbose`:
```
rustc 1.85.0-nightly (21fe748be 2024-12-11)
binary: rustc
commit-hash: 21fe748be15271ea5804e0507cd699b675efe038
commit-date: 2024-12-11
host: x86_64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.5
```

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rust-lang/rust/issues/134221.

The problem was that attributes are "inlined" (backlines are stripped), then when there is an inline comment inside it, the attribute is never considered valid (since unclosed). Fix was to simply put back backlines in case it's a multiline attribute.

r? @notriddle

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Forgot to set only one test thread to prevent having fuzzy sorting...

**Comment by GuillaumeGomez** [comment]
And to replace stdout for time too...

**Comment by cyrgani** [comment]
Does this also fix that 
```rust
//! ```rust
//! #![
//! ```
```
is treated as a sucessful test?

**Comment by GuillaumeGomez** [comment]
Now it does. :)

**Comment by GuillaumeGomez** [comment]
Ready for review.

File: src/librustdoc/doctest/make.rs [hunk 1]
```diff
@@ -520,6 +539,8 @@ fn handle_attr(mod_attr_pending: &mut String, source_info: &mut SourceInfo, edit
         mod_attr_pending.clear();
     } else if mod_attr_pending.ends_with('\\') {
         mod_attr_pending.push('n');
```
Comment (notriddle) [inline src/librustdoc/doctest/make.rs:541]: If I understand this correctly, these lines aren't really needed any more. They would have turned

```rust
#[doc="first\
second"]
```

into

```
#[doc="first\nsecond"]
```

But that only works with double-quoted strings, not raw strings, and is unnecessary now that newlines are preserved here.

```suggestion
```

Comment (GuillaumeGomez) [reply src/librustdoc/doctest/make.rs:541]: Adding a test just in case.

**Comment by GuillaumeGomez** [comment]
Removed the backline handling and also added a regression test for it.

**Comment by GuillaumeGomez** [comment]
Nominating it for beta backporting.

**Pull request merged**